WooCommerce/WooCommerceTests/ViewRelated/Products/Media/MediaPickingCoordinatorTests.swift

```diff
@@ -67,4 +73,24 @@ final class MediaPickingCoordinatorTests: XCTestCase {
         assertEqual(("blaze_edit_ad_form"), eventProperties["flow"] as? String)
     }
 
+    @MainActor
+    func test_showMediaPicker_present_product_image_picker_when_appropriate() {
+        // Given
+        let coordinator = MediaPickingCoordinator(siteID: siteID,
+                                                  imagesOnly: true,
+                                                  allowsMultipleSelections: true,
+                                                  flow: .blazeEditAdForm,
+                                                  analytics: analytics,
+                                                  onCameraCaptureCompletion: { _, _ in },
+                                                  onDeviceMediaLibraryPickerCompletion: { _ in },
+                                                  onWPMediaPickerCompletion: { _ in })
+
+        // When
+        coordinator.showMediaPicker(source: .productMedia(productID: 321), from: sourceViewController)
+
+        // Then
+        waitUntil { [self] in
+            sourceViewController.presentedViewController is ProductImagePickerViewController
+        }
```

Comment:
I think here we should start using [ViewControllerPresentationSpy](https://github.com/woocommerce/woocommerce-ios/pull/13742)

Reply:
That library is necessary only when we want to check for details on the view controllers. Here we are only checking for the correct type of the presented view.